File: frontend-react/src/models/Customer.ts
import Employee from "./Employee";

export default interface Customer {
    id?: string;
    cpf?: string;
    rg?: string;
    email?: string;
    password?: string;
    telephone?: string;
    birthDate?: Date;
    name?: string;
    addressNumber?: number;
    addressZipCode?: string;
    address?: string;
    addressDistrict?: string;
    addressCity?: string;
    addressState?: string;
    employee?: Employee;
}

File: back-end-java/src/main/java/com/org/back_end_java/domain/entity/Customer.java
package com.org.back_end_java.domain.entity;

import java.util.Date;

import jakarta.persistence.Temporal;
import jakarta.persistence.TemporalType;
import lombok.Data;

@Data
public class Customer {
	private String id;
	private String name;
	private String email;
	private String password;
	private String contact;
	private String cpf;
	private String rg;
	@Temporal(TemporalType.DATE)
	private Date dateBirth;
	private Address address;
}
